### PR TITLE
Add namespace property for compatibility with Gradle 8

### LINKED ISSUE #1273: Gradle 8.0 Support
We should add the `namespace` attribute to the build.gradle file.

The sample below was taken from Reid's solution for 1P plugins.

```
android {
    // Conditional for compatibility with AGP <4.2.
    if (project.android.hasProperty("namespace")) {
        namespace 'com.mr.flutter.plugin.filepicker'
    }
}
```

**Describe the bug**
Since Gradle 8.0 the `namespace` attribute was moved from the Manifest.xml to the build.gradle file.

**Platform**

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Desktop

**Details to reproduce the issue**
- Upgrade the example app to Gradle 8.0
- Try to run the app

**Error Log**
```
<omitted part of crash log>

Caused by: java.lang.RuntimeException: Namespace not specified. Please specify a namespace in the module's build.gradle file like so:

android {
    namespace 'com.example.namespace'
}

If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl.getNamespace(VariantDslInfoImpl.kt:77)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl.initApplicationId(VariantDslInfoImpl.kt:128)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl.access$initApplicationId(VariantDslInfoImpl.kt:37)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl$applicationId$2.invoke(VariantDslInfoImpl.kt:94)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl$applicationId$2.invoke(VariantDslInfoImpl.kt:91)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl.getApplicationId(VariantDslInfoImpl.kt:91)
	at com.android.build.gradle.internal.core.dsl.impl.ComponentDslInfoImpl$mergedFlavor$2.invoke(ComponentDslInfoImpl.kt:67)
	at com.android.build.gradle.internal.core.dsl.impl.ComponentDslInfoImpl$mergedFlavor$2.invoke(ComponentDslInfoImpl.kt:63)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.android.build.gradle.internal.core.dsl.impl.ComponentDslInfoImpl.getMergedFlavor(ComponentDslInfoImpl.kt:63)
	at com.android.build.gradle.internal.core.dsl.impl.VariantDslInfoImpl.getMinSdkVersion(VariantDslInfoImpl.kt:63)
	at com.android.build.api.variant.impl.VariantBuilderImpl.<init>(VariantBuilderImpl.kt:48)
	at com.android.build.api.variant.impl.LibraryVariantBuilderImpl.<init>(LibraryVariantBuilderImpl.kt:35)
	at com.android.build.api.variant.impl.LibraryVariantBuilderImpl_Decorated.<init>(Unknown Source)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at org.gradle.internal.instantiation.generator.AsmBackedClassGenerator$InvokeConstructorStrategy.newInstance(AsmBackedClassGenerator.java:1948)
	at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.lambda$newInstance$0(AbstractClassGenerator.java:510)
	at org.gradle.internal.deprecation.DeprecationLogger$4.create(DeprecationLogger.java:295)
	at org.gradle.internal.deprecation.DeprecationLogger.whileDisabledThrowing(DeprecationLogger.java:273)
	at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.newInstance(AbstractClassGenerator.java:509)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:64)
	... 211 more
```

**Flutter Version details**
```
[✓] Flutter (Channel beta, 3.10.0-1.3.pre, on macOS 13.2.1 22D68 darwin-x64, locale en-BE)
    • Flutter version 3.10.0-1.3.pre on channel beta at /Users/navaronbracke/Documents/flutter
    • Upstream repository git@github.com:navaronbracke/flutter.git
    • FLUTTER_GIT_URL = git@github.com:navaronbracke/flutter.git
    • Framework revision d11aff97d2 (2 weeks ago), 2023-04-19 11:11:05 -0700
    • Engine revision a645b36a9b
    • Dart version 3.0.0 (build 3.0.0-417.2.beta)
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/navaronbracke/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /Users/navaronbracke/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.2.1 22D68 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.137

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

**Additional context**
https://docs.flutter.dev/release/breaking-changes/android-java-gradle-migration-guide

### PR DESCRIPTION
**Pull request by navaronbracke**
This PR updates Gradle to version 8.0

Fixes #1273

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -1,2 +1,3 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
```
Comment (navaronbracke) [inline android/gradle.properties:2]: `enableR8` is deprecated, and I got a warning for the missing `useAndroidX`

File: example/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -24,6 +23,7 @@
             android:theme="@style/LaunchTheme"
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density"
             android:hardwareAccelerated="true"
+            android:exported="true"
```
Comment (navaronbracke) [inline example/android/app/src/main/AndroidManifest.xml:26]: This prevented building the example app

File: example/android/build.gradle [hunk 3]
```diff
@@ -30,6 +30,6 @@ subprojects {
     project.evaluationDependsOn(':app')
 }
 
-task clean(type: Delete) {
+tasks.register("clean", Delete) {
```
Comment (navaronbracke) [inline example/android/build.gradle:33]: This is new in newer Gradle versions, was automatically added by the Flutter tool migrators.

File: example/pubspec.yaml [hunk 4]
```diff
@@ -9,8 +9,6 @@ dependencies:
   flutter:
     sdk: flutter
 
-  cupertino_icons: ^0.1.2
```
Comment (navaronbracke) [inline example/pubspec.yaml:12]: `cupertino_icons` was never used and the old version prevented running in null-safe mode.

**Comment by navaronbracke** [comment]
I actually misread the changelog here. Migrating specifically to Gradle 8, requires the source compatibility to change to Java 17, which is not what we want. I'll revert the Gradle 8 change and keep the namespace changes (the latter is what we actually want per https://docs.flutter.dev/release/breaking-changes/android-java-gradle-migration-guide)

**Comment by miguelpruivo** [comment]
Thank you, could you bump it to 5.3.0 instead, since it could break stuff for projects relying on older gradle versions?

**Comment by navaronbracke** [comment]
@miguelpruivo Done!

**Review by miguelpruivo — APPROVED** [review]
LGTM

**Pull request merged**